flaml/automl.py

```diff
@@ -863,7 +863,7 @@ def score(self, X: pd.DataFrame, y: pd.Series, **kwargs):
                 "No estimator is trained. Please run fit with enough budget."
             )
             return None
-        X = self._preprocess(X)
+        X, y = self._preprocess(X, y)
         if self._label_transformer:
             y = self._label_transformer.transform(y)
```

Comment:
line 868 is doing transformation for `y`. There is no need to do transformation for `y` in `_preprocess()` in my opinion.

Reply:
That's a bad design. All label transformation should be done in one place.